### ISSUE TITLE
Google Calendar request slow

### ISSUE THREAD

**Issue by Daanra** [RUGSoftEng/2017-Winter-Summer-School-App#129]
Sometimes when the Google Calendar events are requested from Google, we do not get a response within 10 seconds and receive an error instead. The request should be asynchronous and non-blocking instead so we can happily enjoy our time on the server.

**Comment by Micrified** [comment]
I recall that upon logging onto the server, an initial request would be made to the calendar to fetch the weekdays. If that failed, the page would hang (`/config/dayManipulation.js`).

 However, it *shouldn't* block if it fails to reach Google while the user is paging through the Calendar widget. Can you clarify a bit on the circumstances in which this error happens?

**Comment by Daanra** [comment]
Yeah, sorry the issue was more of a reminder to myself which is why it was not very clear. But to clarify matters, yes if Google Calendar fails to retrieve any items, the server will still render the page. But if it happens, then it will still take a few(?) seconds before the request actually times out. Currently, we require the request to Google to either return something or time out, before we can render the main page. We might want to render the page first, and then send an AJAX request to load all the data. 

But to be honest, this issue should have very low priority at the moment.
